-Wpedantic causes stupid warnings

As noted by @raveit65 here: https://github.com/wmww/gtk-layer-shell/issues/4#issuecomment-511020939
```
[15/28] Compiling C object 'demo/2a97516@@gtk-layer-demo@exe/gtk-layer-demo.c.o'.
../demo/gtk-layer-demo.c:27:21: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
   27 |         .arg_data = (void *)&layer_option_callback,
      |                     ^
../demo/gtk-layer-demo.c:36:21: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
   36 |         .arg_data = (void *)&anchor_option_callback,
      |                     ^
../demo/gtk-layer-demo.c:45:21: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
   45 |         .arg_data = (void *)&margin_option_callback,
      |                     ^
```

--- a/meson.build
+++ b/meson.build
@@ -5,6 +5,10 @@ project('gtk-layer-shell',
     meson_version: '>=0.45.1',
     default_options: ['c_std=c99', 'warning_level=3'])
 
+add_global_arguments(
+    ['-Wno-pedantic'],
+    language: 'c')
+
 gtk = dependency('gtk+-wayland-3.0', version: '>=3.22.0')
 wayland_client = dependency('wayland-client', version: '>=1.10.0')
 # wayland_scanner is required, but we can find it without pkg-config

